src/sensor.cpp

```diff
@@ -1390,221 +1363,27 @@ void log_callback_end( uint32_t fps,
-
-        std::unordered_set<std::shared_ptr<stream_profile_interface>> resolved_req_set;
-        stream_profiles resolved_req;
-        stream_profiles unhandled_reqs(requests);
-
-        // cache the requests
-        for (auto&& req : requests)
-        {
-            _cached_requests[req->get_format()].push_back(req);
-        }
-
-        // while not finished handling all of the requests do
-        while (!unhandled_reqs.empty())
-        {
-            // find the best fitting processing block - the one which resolves the most requests.
-            const auto&& best_match = find_requests_best_pb_match(unhandled_reqs);
-            auto&& best_pbf = best_match.first;
-            auto&& best_reqs = best_match.second;
-
-            // mark as handled resolved requests
-            for (auto&& req : best_reqs)
-            {
-                const auto&& matching_req_predicate = [&req](const std::shared_ptr<stream_profile_interface>& sp) {
-                    return to_profile(req.get()) == to_profile(sp.get());
-                };
-                unhandled_reqs.erase(std::remove_if(begin(unhandled_reqs), end(unhandled_reqs), matching_req_predicate));
-            }
-
-            // Retrieve source profile from cached map and generate the relevant processing block.
-            std::unordered_set<std::shared_ptr<stream_profile_interface>> current_resolved_reqs;
-            auto best_pb = best_pbf->generate();
-            register_processing_block_options(*best_pb);
-            for (auto&& req : best_reqs)
-            {
-                auto&& target = to_profile(req.get());
-                auto&& mapped_source_profiles = _target_to_source_profiles_map[target];
-
-                for (auto&& source_profile : mapped_source_profiles)
-                {
-                    if (best_pbf->has_source(source_profile))
-                    {
-                        resolved_req_set.insert(source_profile);
-                        current_resolved_reqs.insert(source_profile);
-
-                        _profiles_to_processing_block[source_profile].insert(best_pb);
-                    }
-                }
-            }
-            const stream_profiles&& print_current_resolved_reqs = { current_resolved_reqs.begin(), current_resolved_reqs.end() };
-            LOG_INFO("Request: " << best_reqs << "\nResolved to: " << print_current_resolved_reqs);
-        }
-        resolved_req = { resolved_req_set.begin(), resolved_req_set.end() };
-        return resolved_req;
-    }
-
     void synthetic_sensor::open(const stream_profiles& requests)
     {
         std::lock_guard<std::mutex> lock(_synthetic_configure_lock);
 
-        for (auto source : requests)
-            add_source_profile_missing_data(source);
-
-        const auto&& resolved_req = resolve_requests(requests);
+        _formats_converter.prepare_to_convert( requests );
+        
+        const auto & resolved_req = _formats_converter.get_active_source_profiles();
```

Comment:
I wonder if we should add a 'resolve_requests` function that calls this 2 new functions.
IMO `resolve_request` is a well known function in the SDK.
I remember Evgeni always telling me `Try debugging the resolve_request function.`

Reply:
As discussed, no need for this function.